mautic/mautic
# Adding a new boolean option system setting to transliterate p…

…age titles


| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#118
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | fixes #8984 fixes #7525 




#### Description: 
Transliteration of input for UTF8 characters was introduced in Mautic to prevent MySQL from failing when adding Cyrillic languages or emojis.

Now that Mautic 3 ships with UTF8mb4, we can make the Transliteration a "setting" under the Configuration of Mautic.

This new feature does the following:

Introduce a new checkbox option in Mautic Configuration as: "Transliterate page titles"
Add an if statement in PageModel.php to check if the option is on or not.



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Test instructions:

Note: When enabling/disabling the setting please ensure that the cache is cleared.

Case 1: when the Transliterate page titles option is disabled then the page titles in the contacts page will be displayed in their native language, so if the title in Chinese then it will be displayed as-is in Chinese.

Case 1 example:
If the provided title is 你好，世界 then the expected title will be 你好，世界

Case 2: when the Transliterate page titles option is enabled then the page titles in the contacts page will be transliterated to the corresponding spelling in the English language, so if the title in Chinese then it will be transliterated to English.

Case 2 example:
If the provided title is 你好，世界 then the expected title will be ni hao, shi jie

Note: The same example with its values is used in the unit tests.
